### PR TITLE
[Error message] print a better message when allow_multiple is given for non-client mode.

### LINKED ISSUE #29507: [Core] Better error message when using allow_multiple option with a non-ray-client address
### What happened + What you expected to happen

```
import ray
ray.init(address='auto', allow_multiple=True)
Traceback (most recent call last):
  File "/home/david/Projects/theTrading/venvi/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3378, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-f062877bdccd>", line 1, in <module>
    ray.init(address='auto', allow_multiple=True)
  File "/home/david/Projects/theTrading/venvi/lib/python3.9/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/home/david/Projects/theTrading/venvi/lib/python3.9/site-packages/ray/_private/worker.py", line 1262, in init
    raise RuntimeError(f"Unknown keyword argument(s): {unknown}")
RuntimeError: Unknown keyword argument(s): allow_multiple
```
A better error message could be that `allow_multiple` can only be used when the address is in ray-client format.

### Versions / Dependencies

master

### Reproduction script

https://ray-distributed.slack.com/archives/C01DLHZHRBJ/p1666091770165799

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

Instead of 

```
RuntimeError: Unknown keyword argument(s): allow_multiple
```

It prints
```
RuntimeError: `allow_multiple` argument is passed to `ray.init` when the ray client is not used (https://docs.ray.io/en/master/cluster/running-applications/job-submission/ray-client.html#connect-to-multiple-ray-clusters-experimental). Do not pass the `allow_multiple` to `ray.init` to fix the issue.
```

## Related issue number

Closes https://github.com/ray-project/ray/issues/29507

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -1263,6 +1263,15 @@ def init(
         usage_lib.record_library_usage("client")
         return ctx
 
+    if kwargs.get("allow_multiple"):
+        raise RuntimeError(
+            "`allow_multiple` argument is passed to `ray.init` when the "
+            "ray client is not used ("
+            "https://docs.ray.io/en/master/cluster/running-applications/job-submission"
```
Comment (jjyao) [inline python/ray/_private/worker.py:1270]: Feel we need to find a way to link check those links as well.

**Review by jjyao — APPROVED** [review]

File: python/ray/_private/worker.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/_private/worker.py:1270]: how do we do this now?

Comment (jjyao) [inline python/ray/_private/worker.py:1270]: Actually can you replace master with `{get_ray_doc_version()}` to make sure the link will always be valid?

Comment (rkooo567) [reply python/ray/_private/worker.py]: Followed up from the slack. Let's do this in a separate PR?

**Pull request merged**